codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttEncoder.java

```diff
@@ -325,11 +418,53 @@ private static ByteBuf encodePublishMessage(
         if (mqttFixedHeader.qosLevel().value() > 0) {
             buf.writeShort(variableHeader.packetId());
         }
+        buf.writeBytes(propertiesBuf);
         buf.writeBytes(payload);
 
         return buf;
     }
 
+    private static ByteBuf encodePubReplyMessage(ChannelHandlerContext ctx,
+                                          ByteBufAllocator byteBufAllocator,
+                                          MqttMessage message) {
+        if (message.variableHeader() instanceof MqttPubReplyMessageVariableHeader) {
+            MqttFixedHeader mqttFixedHeader = message.fixedHeader();
+            MqttPubReplyMessageVariableHeader variableHeader =
+                    (MqttPubReplyMessageVariableHeader) message.variableHeader();
+            int msgId = variableHeader.messageId();
+
+            final ByteBuf propertiesBuf;
```

Comment:
you will need to call `propertiesBuf.release()` later as you not transfer ownership